content/en/blog/2023/ambient-performance-savings/index.md

```diff
@@ -0,0 +1,133 @@
+---
+title: "Cut Service Mesh Overhead with Istio Ambient Mesh"
+description: An anaylsis of how cloud resources are utilized in ambient and sidecar service mesh architectures.
+publishdate: 2023-07-14
+attribution: "Greg Hanson - Solo.io"
+keywords: [istio,ambient,waypoint,ztunnel]
+---
+
+Istio Ambient Mesh is the new Istio data plane without requiring sidecars, refer to the [announcement](/blog/2022/introducing-ambient-mesh/) for more details. Istio Ambient Mesh is included in [Istio 1.18](https://github.com/istio/istio/releases/tag/1.18.0) but is still considered to be in [alpha feature status](/docs/releases/feature-stages/#feature-phase-definitions).
+
+Sidecars have been a staple of Istio’s architecture since day one and are responsible for the majority of features available in Istio today. However, sidecars require the injection of an additional container to each Kubernetes pod resource, each of which needs allocated resources from the pod.
+
+Many of you are familiar with the simplified operation brought by the Istio ambient architecture. Let’s explore how ambient mesh cuts the service mesh costs typically associated with sidecars.
+
+## Check Out the Savings
+
+Our test scenario deploys four instances of the [Online Boutique](https://github.com/GoogleCloudPlatform/microservices-demo), each to their own namespace, with each underlying microservice scaled to 3 replicas. There will be a total of 48 application services and 144 applications pods in this setup. Test scenarios utilize the provided [loadgenerator](https://github.com/GoogleCloudPlatform/microservices-demo/tree/main/src/loadgenerator) microservice running in each of the namespaces to generate traffic during the observed window. The tests deployed Istio in several scenarios and resource usage and allocation are compared across the runs. The scenarios are:
```

Comment:
You're currently running a demo that utilizes a loadgenerator to simulate actual traffic. I'm interested in determining the volume of traffic and the QPS (queries per second) rate, as this information is crucial for analyzing the resource utilization of various modes.

Reply:
here is the source code for loadgenerator, it cycles through a list of different operations (lookup, browse, checkout, etc):
https://github.com/GoogleCloudPlatform/microservices-demo/blob/main/src/loadgenerator/locustfile.py#L67

There is an open issue in that repo for exposing the ability to tune things like request freuqency, but it currently isn't supported:
https://github.com/GoogleCloudPlatform/microservices-demo/issues/1692

the current QPS of each loadgenerator service is about 1-2 seconds for each request.

This was something I struggled with.  Bookinfo could not handle a large amount of requests per second, but I couldn't find a public microservice demo that could handle something like fortio driving large QPS.  The other part I struggled with was then writing an appropriate driver.  Something like fortio definitely gave me the options for configuring QPS, but to mimic a true env I would need it to be hitting a list of API's rather than a single one for the entire test duration.  These were the factors that led me to choose the boutique demo

Reply:
That make sense. Have you take the [distributed load generator](https://docs.locust.io/en/stable/running-distributed.html) of Locust into consideration?